### PR TITLE
MGMT-15824: Gather networking info

### PR DESCRIPTION
**Pull request by tsorya**
Gathering network configuration files on PreUpgrade
Currently gathering paths:
1. hostname
2. system-connections
3. ovn chosen interface in order to know which interface is the default

### CODE REVIEW

File: controllers/upgrade_handlers.go [hunk 1]
```diff
@@ -30,7 +30,11 @@ func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *ra
 	// pre upgrade steps
 	if err := r.ClusterConfig.FetchClusterConfig(ctx); err != nil {
```
Comment (jc-rh) [inline controllers/upgrade_handlers.go:31]: @donpenney Do you think the prep handler can make use of these or we really only need them for the upgrade stage?

Comment (browsell) [reply controllers/upgrade_handlers.go:31]: Generally we want to collect the data as close to the upgrade as possible, therefore we collect during the upgrade stage. The prep conceivably could be days before the actual upgrade

**Comment by romfreiman** [comment]
What's the process? who approves it?

**Review by donpenney — APPROVED** [review]

**Pull request merged**